lib/beaker/hypervisor/vmpooler.rb

```diff
@@ -15,6 +15,8 @@ class Vmpooler < Beaker::Hypervisor
       Errno::ENETUNREACH,
     ]
 
+    attr_reader :options, :logger, :hosts, :credentials
```

Comment:
any reason that you need these when those variables are already stored as @options/@logger/etc?